Test failure in TestStringImageFileWithText since 16.1.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Ever since version 16.1.0, at least one test fails on NixOS when building the package from source. This could obviously be an error caused by the way NixOS does things, but since I can't figure it out myself I'm posting the stack trace here. Hope you can help solve this!

### Theme

n/a

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
running tests
ok      github.com/jandedobbeleer/oh-my-posh/src        0.004s [no tests to run]
ok      github.com/jandedobbeleer/oh-my-posh/src/ansi   0.003s
--- FAIL: TestStringImageFileWithText (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xb3369d]

goroutine 9 [running]:
testing.tRunner.func1.2({0xc2e3a0, 0x13505d0})
        /nix/store/8v5zwymidmry0wd3lhj6zggskzsvqrfk-go-1.20.4/share/go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        /nix/store/8v5zwymidmry0wd3lhj6zggskzsvqrfk-go-1.20.4/share/go/src/testing/testing.go:1529 +0x39f
panic({0xc2e3a0, 0x13505d0})
        /nix/store/8v5zwymidmry0wd3lhj6zggskzsvqrfk-go-1.20.4/share/go/src/runtime/panic.go:884 +0x213
golang.org/x/image/font.MeasureString({0x0, 0x0}, {0xc00052c5a0, 0xc})
        /build/source/src/vendor/golang.org/x/image/font/font.go:288 +0x3d
golang.org/x/image/font.(*Drawer).MeasureString(...)
        /build/source/src/vendor/golang.org/x/image/font/font.go:216
github.com/jandedobbeleer/oh-my-posh/src/engine.(*ImageRenderer).measureContent(0xc0004f9d80)
        /build/source/src/engine/image.go:398 +0xb8
github.com/jandedobbeleer/oh-my-posh/src/engine.(*ImageRenderer).SavePNG(0xc0004f9d80)
        /build/source/src/engine/image.go:414 +0x46
github.com/jandedobbeleer/oh-my-posh/src/engine.runImageTest({0xcfee3a, 0x19}, {0xceaa88, 0x6})
        /build/source/src/engine/image_test.go:52 +0x1d3
github.com/jandedobbeleer/oh-my-posh/src/engine.TestStringImageFileWithText(0x0?)
        /build/source/src/engine/image_test.go:61 +0x7f
testing.tRunner(0xc0003e6340, 0xd47c78)
        /nix/store/8v5zwymidmry0wd3lhj6zggskzsvqrfk-go-1.20.4/share/go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        /nix/store/8v5zwymidmry0wd3lhj6zggskzsvqrfk-go-1.20.4/share/go/src/testing/testing.go:1629 +0x3ea
FAIL    github.com/jandedobbeleer/oh-my-posh/src/engine 0.010s
FAIL
```

--- a/src/engine/config.go
+++ b/src/engine/config.go
@@ -270,8 +270,12 @@ func escapeGlyphs(s string, migrate bool) string {
 	}
 
 	var cp codePoints
+	var err error
 	if migrate {
-		cp = getGlyphCodePoints()
+		cp, err = getGlyphCodePoints()
+		if err != nil {
+			migrate = false
+		}
 	}
 
 	var builder strings.Builder

--- a/src/engine/image.go
+++ b/src/engine/image.go
@@ -45,6 +45,14 @@ import (
 	"golang.org/x/image/font"
 )
 
+type ConnectionError struct {
+	reason string
+}
+
+func (f *ConnectionError) Error() string {
+	return f.reason
+}
+
 const (
 	red    = "#ED655A"
 	yellow = "#E1C04C"
@@ -150,7 +158,7 @@ func (ir *ImageRenderer) Init(env platform.Environment) error {
 	ir.cleanContent()
 
 	if err := ir.loadFonts(); err != nil {
-		return err
+		return &ConnectionError{reason: err.Error()}
 	}
 
 	ir.defaultForegroundColor = &RGB{255, 255, 255}

--- a/src/engine/image_test.go
+++ b/src/engine/image_test.go
@@ -48,17 +48,26 @@ func runImageTest(config, content string) (string, error) {
 		},
 	}
 
-	_ = image.Init(env)
+	err = image.Init(env)
+	if err != nil {
+		return "", err
+	}
+
 	err = image.SavePNG()
 	if err == nil {
 		os.Remove(image.Path)
 	}
+
 	return filepath.Base(image.Path), err
 }
 
 func TestStringImageFileWithText(t *testing.T) {
 	for _, tc := range cases {
 		filename, err := runImageTest(tc.Config, "foobar")
+		if connectionError, ok := err.(*ConnectionError); ok {
+			t.Log(connectionError.Error())
+			continue
+		}
 		assert.Equal(t, "jandedobbeleer.png", filename, tc.Case)
 		assert.NoError(t, err)
 	}
@@ -68,6 +77,10 @@ func TestStringImageFileWithANSI(t *testing.T) {
 	prompt := `[38;2;40;105;131m[0m[48;2;40;105;131m[38;2;224;222;244m jan [0m[38;2;40;105;131m[0m[38;2;224;222;244m [0m`
 	for _, tc := range cases {
 		filename, err := runImageTest(tc.Config, prompt)
+		if connectionError, ok := err.(*ConnectionError); ok {
+			t.Log(connectionError.Error())
+			continue
+		}
 		assert.Equal(t, "jandedobbeleer.png", filename, tc.Case)
 		assert.NoError(t, err)
 	}

--- a/src/engine/migrate_glyphs.go
+++ b/src/engine/migrate_glyphs.go
@@ -10,7 +10,7 @@ import (
 
 type codePoints map[int]int
 
-func getGlyphCodePoints() codePoints {
+func getGlyphCodePoints() (codePoints, error) {
 	var codePoints = make(codePoints)
 
 	client := &http.Client{}
@@ -19,19 +19,19 @@ func getGlyphCodePoints() codePoints {
 
 	request, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://ohmyposh.dev/codepoints.csv", nil)
 	if err != nil {
-		return codePoints
+		return codePoints, &ConnectionError{reason: err.Error()}
 	}
 
 	response, err := client.Do(request)
 	if err != nil {
-		return codePoints
+		return codePoints, err
 	}
 
 	defer response.Body.Close()
 
 	lines, err := csv.NewReader(response.Body).ReadAll()
 	if err != nil {
-		return codePoints
+		return codePoints, err
 	}
 
 	for _, line := range lines {
@@ -48,5 +48,5 @@ func getGlyphCodePoints() codePoints {
 		}
 		codePoints[int(oldGlyph)] = int(newGlyph)
 	}
-	return codePoints
+	return codePoints, nil
 }

--- a/src/engine/migrate_glyphs_test.go
+++ b/src/engine/migrate_glyphs_test.go
@@ -7,6 +7,10 @@ import (
 )
 
 func TestGetCodePoints(t *testing.T) {
-	codepoints := getGlyphCodePoints()
+	codepoints, err := getGlyphCodePoints()
+	if connectionError, ok := err.(*ConnectionError); ok {
+		t.Log(connectionError.Error())
+		return
+	}
 	assert.Equal(t, 1939, len(codepoints))
 }

